### PR TITLE
Add AlphaFilter for 3D rotations

### PR DESCRIPTION
**Pull request by bresch**
### Solved Problem
Filtering angles require special care as they live on a sphere.

### Solution
Create a template specialization to run the filter update on SO(3) using quaternions.

### Changelog Entry
```
-
```

### Test coverage
- Unit tests
- EKF2 replay

@MaEtUgR We could extend it a bit more to allow for a `Vector3` "alpha" and use it for attitude input filtering
We could also do the same for a 2D rotation (specialization for Dcm2) in order do handle angle wrapping properly.

Note that the change indicator diff is because I modified the update equation from 
`y = (1-a)*y + a*u` to `y = y + a*(u-y)` to look like the one used for the quaternion filter (where it actually needs to have this form).

### CODE REVIEW

File: src/lib/mathlib/math/filter/AlphaFilter.hpp [hunk 1]
```diff
@@ -116,9 +116,40 @@ class AlphaFilter
 	float getCutoffFreq() const { return _cutoff_freq; }
 
 protected:
-	T updateCalculation(const T &sample) { return (1.f - _alpha) * _filter_state + _alpha * sample; }
+	virtual T updateCalculation(const T &sample) = 0;
```
[the remark this answers is not in the document]
Comment (dagar) [reply src/lib/mathlib/math/filter/AlphaFilter.hpp:119]: The virtual dispatch (vtable) bloats every single usage.

image

[the remark this answers is not in the document]
Comment (bresch) [reply src/lib/mathlib/math/filter/AlphaFilter.hpp]: Alright, I changed it to use function overload instead.

File: src/lib/mathlib/math/test/AlphaFilterTest.cpp [hunk 2]
```diff
@@ -280,3 +282,51 @@ TEST(AlphaFilterTest, ConvergenceTest)
 
 	EXPECT_NEAR(last_value, -100.f, 1e-4f);
 }
+
+TEST(AlphaFilterTest, runPositiveRot3)
+{
+	// GIVEN an input of 1 in a filter with a default time constant of 9 (alpha = 0.9)
```
Comment (MaEtUgR) [inline src/lib/mathlib/math/test/AlphaFilterTest.cpp:288]:
```suggestion
	// GIVEN an input with sample interval 1 in a filter with a time constant of 9 (alpha = 0.1)
```
Or did I understand this wrong?

**Review by MaEtUgR — APPROVED** [review]

File: src/lib/mathlib/math/test/AlphaFilterTest.cpp (hunk 2 above)
Comment (bresch) [reply src/lib/mathlib/math/test/AlphaFilterTest.cpp:288]: no, you're right, I blindly copy-pasted another wrong comment...

**Pull request merged**